core/src/Kokkos_Core_fwd.hpp

```diff
@@ -249,6 +124,19 @@ struct VerifyExecutionCanAccessMemorySpace<Space, Space> {
   KOKKOS_INLINE_FUNCTION static void verify(void) {}
   KOKKOS_INLINE_FUNCTION static void verify(const void *) {}
 };
+
+// Base class for exec space initializer factories
+class ExecSpaceFactoryBase {
```

Comment:
Yeah, it's not really a factory.  I just changed it to ExecSpaceInitializerBase ( and derived classes ###SpaceInitializer ), but since it has a finalize and fence method, maybe it should be called something else.  Perhaps ExecSpaceControlBase?